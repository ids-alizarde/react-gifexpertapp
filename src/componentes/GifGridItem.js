import React from 'react'

export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__rubberBand">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
