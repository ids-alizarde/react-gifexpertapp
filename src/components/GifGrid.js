import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../services/GetGifs';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ]);

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeInTopLeft">{ category }</h3>

            { loading && <p>Loading.........</p> }

            <div className="card-grid">

                { images.map( img => (

                    <GifGridItem 
                        key={img.id}
                        { ...img }>

                    </GifGridItem>

                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
