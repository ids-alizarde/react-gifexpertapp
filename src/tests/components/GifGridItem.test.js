import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {

    const title = 'Un titulo X';
    const url = 'https://localhost/al.com.jpg'
    const wrapper = shallow( <GifGridItem title = { title } url = { url }/> );
    
    test('Debe mostrar correctamente el componente <GifGridItem/>', () => {

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de tener un parrafo con el texto', () => {
        
        const parrafo = wrapper.find( 'p' );
        const text = parrafo.text().trim();
        expect( text ).toBe( title );
    });

    test('Debe de mostrar la imagen igual al url y alt de los props', () => {

        const img = wrapper.find( 'img' );
        // console.log( img.html() );
        // console.log( img.props() );
        // console.log( img.props().src );
        // console.log( img.prop( 'src' ) );
        expect( img.props().src ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );

    });

    // Forma 1
    test('Debe de tener la clase animate__rubberBand de Animate.css 1', () => {
        
        const div = wrapper.find( 'div' );
        expect( div.hasClass( 'animate__rubberBand' ) ).toBe( true );

    });

    // Forma 2
    test('Debe de tener la clase animate__rubberBand de Animate.css 2', () => {
        
        const div = wrapper.find( 'div' );
        const className = div.prop( 'className' );
        expect( className.includes( 'animate__rubberBand' ) ).toBe( true );

    });
    
});
