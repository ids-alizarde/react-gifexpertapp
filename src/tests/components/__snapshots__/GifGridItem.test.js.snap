// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar correctamente el componente <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Un titulo X"
    src="https://localhost/al.com.jpg"
  />
  <p>
    Un titulo X
  </p>
</div>
`;
